src/action/finder/download.rs

```diff
@@ -0,0 +1,287 @@
+use crate::action::finder::chunks::{get_file_chunks, GetFileChunksConfig, Chunks};
+use crate::action::finder::request::{
+    DownloadActionOptions, HashActionOptions,
+};
+use crate::fs::Entry;
+use rrg_proto::file_finder_download_action_options::OversizedFilePolicy as DownloadOversizedFilePolicy;
+use rrg_proto::file_finder_hash_action_options::OversizedFilePolicy as HashOversizedFilePolicy;
+use sha2::{Digest, Sha256};
+use std::fs::File;
+use std::io::{BufReader, Take};
+
+#[derive(Debug)]
+pub enum Response {
+    /// Download action is not performed and no further action is required.
+    Skip(),
+    /// File was not downloaded, but hash action must be executed.
+    HashRequest(HashActionOptions),
+    /// Chunks of data to be downloaded.
+    DownloadData(Chunks<BufReader<Take<File>>>),
```

Comment:
I would go with "collect" nomenclature rather than "download" (because technically what the agent does is actually "uploading"...).

Reply:
Changed to `CollectData`. Changed the comment to `/// Chunks of data to be uploaded.`.